addons/web/static/src/legacy/legacy_views.js

```diff
@@ -126,6 +319,28 @@ function registerView(name, LegacyView) {
                 views,
             });
         }
+
+        exportState() {
+            const state = this.controllerRef.comp.exportState();
+            delete state.searchModel;
+            delete state.searchPanel;
+            return state;
+        }
+
+        share() {
```

Comment:
not a fan of the function name. In legacy, `exportState` was `getLocalState`. Maybe we could have a similar terminology, like `exportLocalState` and `exportGlobalState` or something.

Reply:
to discuss with team

Reply:
discuss exportLocalState, exportGlobalState and related stuff.
I have put resIds in globalState